### PR TITLE
Initial testbed Architecture

### PR DESCRIPTION
**Pull request by joshmedeski**

### CODE REVIEW

**Comment by joshmedeski** [comment]
@djscheuf @chapmanj, this is a pull request.

The idea is to package together commits from a branch and submit a request to merge this code into the master branch after completing a feature. 

This is a great place to review the code, talk about the features, and potential suggest changes and add more commits to complete the request.

I'll review this code this week to get the ball rolling!

File: project/.travis.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
+# Continuous Integration (CI) is the practice, in software
```
Comment (joshmedeski) [inline project/.travis.yml:1]: Why was this file added to the project? Do you plan on using Travis?

**Comment by joshmedeski** [comment]
I went over the files. Still don't know much of what I'm looking at so it's hard for me to give any valuable feedback.

File: project/.travis.yml (hunk 1 above)
Comment (djscheuf) [inline project/.travis.yml:1]: To the best of my knowledge it was added during the Platformio Init. You could try deleting it and seeing if it has any effect on the system, as it is possible that it was added from some other package that I was using.

**Comment by djscheuf** [comment]
Hm... Let's try this then. I had two goals with this branch. 1. I wanted to organize our files in a manner that would make it easy for us to both develop and to test our code. 
And secondly I wasnted to support any operating system we could be using for development. 

As a result, I chose to use Atom as an IDE for the following reasons. In the first place, Atom is cross platform, and has wide support for many functions, which are universally available on Atom no matter the OS. For a second, Atom can serve as an IDE for Arduino, but using the launguae-arduino and the Platformio packages.

Platformio serves as the compiler, builder, and flasher for the arduino code, and might ( haven't checked) allow for Serial port feedback. However it requires a certain base hierarchy for its projects. This can be seen demonstrated in the current state of the branch.

The Arduino IDE, on the other hand, supports multiple platforms, but makes a mockery of project organization, generally requiring file to be present under the root folder, and no deeper. So while the support might be better, Atom is more flexible and acheives a better segregation of the code.

If you look through the set up of the code, using the Atom IDE, and the specified packages; evaluate the code based on your comfort using this tool. If you find it workable, then there is little reason to keep it out of master. If you find your are troubled, then that is a subject for discussion.

Does this help? (The code, I expect should be mostly self-explanatory).

**Comment by djscheuf** [comment]
Some additional information: I was able to confirm that my Blinker code ( as written in the main.cpp of the pull request) Does function as expected. I have not yet been able to determine its effects on the timing of other functions, such as delay(). However I believe the working test is sufficient for the moment.

Separately, I realized that I should probably specify that in order to build and run with the Platformio package, you will need to open the project folder directly, not the Arduino Hackathon. This is because Platformio will be unable to identify the necessary .ini file if you we use the ArduinoHackathon folder( since it was not initialized). However THe project folder has been thus-ly initialized and will successfully build.

**Comment by joshmedeski** [comment]
Everything looks good for now. :+1:
